src/Products/Formulator/EmailLinesField.py

```diff
@@ -15,7 +16,10 @@
 # needed), and of course no characters that don't belong in an e-mail
 # address.
 
-pattern = re.compile('^[0-9a-zA-Z_&.%+-]+@([0-9a-zA-Z]([0-9a-zA-Z-]*[0-9a-zA-Z])?\.)+[a-zA-Z]{2,6}$')
+pattern = re.compile(
+    '^[0-9a-zA-Z_&.%+-]+@([0-9a-zA-Z]([0-9a-zA-Z-]*[0-9a-zA-Z])?\\.)'
```

Comment:
There is an addition of `\` here at `?\\.` Was that intended? It seems reasonable, otherwise a `r''` might be an option.